packages/http/src/validator/index.ts

```diff
@@ -2,6 +2,7 @@ import { IPrismDiagnostic, ValidatorFn } from '@stoplight/prism-core';
 import { DiagnosticSeverity, IHttpOperation } from '@stoplight/types';
 import * as caseless from 'caseless';
 
+import { toUndefined } from 'fp-ts/lib/Option';
```

Comment:
Ok here's something to keep in mind; in general if you're going to a Monad (`Option, Either, Task, TaskEither`) you should "never" go back to the "bad world". `toUndefined` is a method effectively bringing back in the uncertainty and it should be avoided.

The how depends on the context and the code. Follow this review, and we'll figure out this specific one.